### PR TITLE
docs: Add more detail around release process

### LINKED ISSUE #778: Document release complete process
**Is your feature request related to a problem? Please describe.**
As a new user, it’s unclear how to actually create a release. For instance:

1. when merging a release PR, what merge strategy should be used? My projects generally use squashed merges, but that clearly won’t work correctly with a release PR. What is the expected merge strategy?
2. Make it clear that we have to manually tag releases ourselves. This surprised me and isn’t very clear from the README.
3. How do I go from a release PR to a GitHub release? Am I expected to manually create one?
4. Is it expected that releases are always cut from the main branch?

 **Describe the solution you'd like**

Update the README to describe how to do a release.

### PR DESCRIPTION
**Pull request by nzakas**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/release-please/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #778 🦕

I tried updating the README with the answers from #778. I hope this helps. Thanks so much for release-please!

### CODE REVIEW

**Comment by nzakas** [comment]
@googlebot I signed it!

File: README.md [hunk 1]
```diff
@@ -20,7 +20,21 @@ release-please maintains Release PRs:
...
+1. Updates your changelog file (for example `CHANGELOG.md`)
+2. Tags the commit with the version number
+3. Creates a GitHub Release based on the tag
+
+You can tell where the Release PR is its lifecycle by the status label on the
+PR itself:
+
+- `autorelease:pending` is the initial state of the Release PR before it is merged
+- `autorelease:tagged` means that the Release PR has been merged and the release has been tagged in GitHub
+- `autorelease:published` means that a GitHub release has been published based on the Release PR
```
Comment (chingor13) [inline README.md:37]: Let's remove this last label entry -- the actual publishing of packages is outside the scope of release-please and this is a convention other tools our team uses. Alternatively, we might mention that this is a suggestion for your choice of publishing automation to adopt.

**Review by chingor13 — COMMENTED** [review]
Thanks for this!

File: README.md (hunk 1 above)
Comment (chingor13) [reply README.md:37]: Eh, this is a nit, and we can fix this on our side.

**Review by chingor13 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (bcoe) [inline README.md:37]:
```suggestion
- `autorelease:published` means that a GitHub release has been published based on the Release PR (_release-please does not automatically add this tag, but we recommend it as a convention for publication tooling_).
```

File: README.md [hunk 2]
```diff
@@ -20,7 +20,21 @@ release-please maintains Release PRs:
 <img width="400" src="/screen.png">
 
 These Release PRs are kept up-to-date as additional work is merged. When you're
-ready to tag a release, simply merge the release PR.
+ready to tag a release, simply merge the release PR. Both squash-merge and
+merge commits work with Release PRs.
+
+When the Release PR is merged, release-please takes the following steps:
+
+1. Updates your changelog file (for example `CHANGELOG.md`)
```
Comment (bcoe) [inline README.md:28]:
```suggestion
1. Updates your changelog file (for example `CHANGELOG.md`), along with other language specific files (for example `package.json`).
```

**Review by bcoe — CHANGES_REQUESTED** [review]
Thanks for the contribution 👏 

Added a couple suggestions, let me know what you think.

**Comment by nzakas** [comment]
LGTM. Thanks!

**Review by bcoe — APPROVED** [review]

**Comment by bcoe** [comment]
@nzakas thank you for the contribution 👍

**Pull request merged**